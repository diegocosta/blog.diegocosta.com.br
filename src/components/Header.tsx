import React from 'react';
import styled from 'styled-components';
import { Link } from 'gatsby';

interface HeaderProps {
  small?: boolean;
}

const Container = styled.header`
  margin-bottom: 20px;
`;

const Name = styled.span<HeaderProps>`
  font-size: ${props => props.small ? '6vw' : '15vw'};
  color: #fff;
  font-weight: 700;

  @media (min-width: 760px) {
    font-size: ${props => props.small ? '24px' : '80px'};
  }
`;

const LastName = styled.span`
  color: #d73738;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;
`;

function Header(props: HeaderProps) {
  return (
    <Container>
      <StyledLink to={`/`}>
        <Name small={props.small}>diego<LastName>c.</LastName></Name>
      </StyledLink>
    </Container>
  );
}

Header.defaultProps = {
  small: false,
};

export default Header;