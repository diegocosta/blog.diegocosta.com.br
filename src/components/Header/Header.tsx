import React from 'react';
import styled from 'styled-components';
import { Link } from 'gatsby';
import { FaChevronLeft } from 'react-icons/fa';

import Navigation from '~/components/Navigation';
import ThemeSwitcher from '~/components/ThemeSwitcher';

const Container = styled.header`
  margin-bottom: 40px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  nav {
    margin-top: 10px;
    display: none;

    @media (min-width: 760px) {
      display: block;
    }
  }
`;

const Name = styled.span<{ small: boolean }>`
  font-size: ${(props) => (props.small ? '8vw' : '15vw')};
  color: ${({ theme }) => theme.titleColor};
  font-weight: 700;
  text-transform: lowercase;

  @media (min-width: 760px) {
    font-size: ${(props) => (props.small ? '34px' : '80px')};
  }
`;

const LastName = styled.span`
  color: ${({ theme }) => theme.accentColor};
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;

  :focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

const Options = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const BackButton = styled(FaChevronLeft)`
  color: ${({ theme }) => theme.titleColor};
  font-size: 22px;
  font-weight: bold;
  margin-right: 10px;
`;

interface HeaderProps {
  small: boolean;
  author: string;
}

export default function Header(props: HeaderProps): React.ReactElement {
  const [name, lastname] = props.author.split(' ');

  return (
    <Container>
      <StyledLink to="/">
        {props.small && <BackButton />}
        <Name small={props.small}>
          {name}
          {lastname && <LastName data-testid="header-lastname">{lastname[0]}.</LastName>}
        </Name>
      </StyledLink>
      <Options>
        <Navigation />
        <ThemeSwitcher />
      </Options>
    </Container>
  );
}
