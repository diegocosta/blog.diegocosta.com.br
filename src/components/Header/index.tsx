import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Header from './Header';

export default ({ small }: { small: boolean }) => {
  const {
    site: {
      siteMetadata: { author },
    },
  } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            author
          }
        }
      }
    `
  );

  return <Header small={small} author={author} />;
};
