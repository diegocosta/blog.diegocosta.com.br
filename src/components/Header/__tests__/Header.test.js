import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';

jest.mock('../../Navigation', () => () => <p>Navigation</p>);

describe('<Header />', () => {
  it('should render properly', () => {
    const { baseElement, getByTestId, getByText } = render(<Header small={false} author="John Doe" />);
    expect(getByTestId('header-lastname')).toBeTruthy();
    expect(getByText('John')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it('should not render the last name', () => {
    const { baseElement, queryByTestId } = render(<Header small={true} author="John" />);
    expect(queryByTestId('header-lastname')).toBeFalsy();
    expect(baseElement).toMatchSnapshot();
  });
});
