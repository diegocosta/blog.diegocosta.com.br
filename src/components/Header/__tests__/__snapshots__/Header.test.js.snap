// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should not render the last name 1`] = `
<body>
  .c0 {
  margin-bottom: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 nav {
  margin-top: 10px;
  display: none;
}

.c3 {
  font-size: 8vw;
  font-weight: 700;
  text-transform: lowercase;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
}

.c1:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 22px;
  font-weight: bold;
  margin-right: 10px;
}

@media (min-width:760px) {
  .c0 nav {
    display: block;
  }
}

@media (min-width:760px) {
  .c3 {
    font-size: 34px;
  }
}

<div>
    <header
      class="c0"
    >
      <a
        class="c1"
        href="/"
      >
        <svg
          class="c2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
        <span
          class="c3"
        >
          John
        </span>
      </a>
      <div
        class="c4"
      >
        <p>
          Navigation
        </p>
        <p>
          ThemeSwitcher
        </p>
      </div>
    </header>
  </div>
</body>
`;

exports[`<Header /> should render properly 1`] = `
.c0 {
  margin-bottom: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 nav {
  margin-top: 10px;
  display: none;
}

.c2 {
  font-size: 15vw;
  font-weight: 700;
  text-transform: lowercase;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
}

.c1:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:760px) {
  .c0 nav {
    display: block;
  }
}

@media (min-width:760px) {
  .c2 {
    font-size: 80px;
  }
}

<body>
  <div>
    <header
      class="c0"
    >
      <a
        class="c1"
        href="/"
      >
        <span
          class="c2"
        >
          John
          <span
            class=""
            data-testid="header-lastname"
          >
            D
            .
          </span>
        </span>
      </a>
      <div
        class="c3"
      >
        <p>
          Navigation
        </p>
        <p>
          ThemeSwitcher
        </p>
      </div>
    </header>
  </div>
</body>
`;
